app/views/posts/_post.html.erb

```diff
@@ -0,0 +1,13 @@
+<% @topic.posts.each do |post| %>
```

Comment:
This is a great step in the right direction, this actually doesn't support as much code reuse as it code
1. you are still use the `@topic` variable from the controller, so it is tired to only be used in controller actions which define `@topic` 
2. You are looping through multiple `posts` so that can't be used in places where you only have one post (like `show`)

Instead, I generally have my partials represent a single `post`.  Then you can loop through the collection elsewhere and call the partial in a loop, passing an individual local `post` variable to a partial on each loop iteration.

Here's an example of passing a local variable to a partial
`<%= render partial: 'comments/comment', locals: {comment: comment} %>`